### ISSUE TITLE
InfluxDB: maxdatapoints should be per series

### ISSUE THREAD

**Issue by alex88** [grafana/grafana#26431]
This is a follow-up of https://github.com/grafana/influxdb-flux-datasource/issues/116

**What happened**:
When trying to run this query (which works fine on the influxdb query editor):

```
import "date"

from(bucket: "my-bucket")
  |> range(start: date.truncate(t: time(v: $__from * 1000000), unit: 5m), stop: date.truncate(t: time(v: $__to * 1000000), unit: 5m))
  |> filter(fn: (r) =>
    r._measurement == "metric"
    and r.machine == "83"
    and (r._field == "temp_1"
    or r._field == "temp_2"
    or r._field == "temp_3"
    or r._field == "temp_4"
    or r._field == "temp_5"
    or r._field == "temp_6")
  )
  |> aggregateWindow(every: 5m, fn: mean)
```

if the dashboard time interval is 7 days or more (in our case), the table only shows temp_1 and the last timestamp is a few hours behind the real data shown in the influxdb editor. I can see in the ajax request that the server returns only 1 dataframe.
If I then open the graph panel and click on refresh it returns all the 5 dataframes with all the data.
If I refresh the whole page instead, the problem persists.
There are no error logs on the grafana container nor the influxdb server.

**What you expected to happen**:
Show all the data the first time the query is executed

**How to reproduce it (as minimally and precisely as possible)**:
I'm not sure, probably run a query that returns an high enough amount of data

**Environment**:
- Grafana version: 7.1.0
- Data source type & version: Influxdb (flux language)
- OS Grafana is installed on: official 7.1.0 docker image
- User OS & Browser: OSX - Chrome
- Grafana plugins: None

**Comment by ryantxu** [comment]
@alex88 can you attach the csv results you get when you run this in chronograph?

Real test data has helped LOTS for catching these edge cases:
https://github.com/grafana/grafana/tree/master/pkg/tsdb/influxdb/flux/testdata

**Comment by alex88** [comment]
@ryantxu what's cronograph? forgot to mention server is influxdb 2.0-beta

**Comment by alex88** [comment]
This is the CSV got from the query editor in the influxdb dashboard

[2020-07-17-16-32_chronograf_data.csv.zip](https://github.com/grafana/grafana/files/4940715/2020-07-17-16-32_chronograf_data.csv.zip)

**Comment by alex88** [comment]
Just to better show what I see:

I load the dasboard and the panel only shows temp_1
![Screen Shot 2020-07-17 at 4 34 44 PM](https://user-images.githubusercontent.com/436888/87838345-9c621f00-c84b-11ea-917d-d4a4c6de2d96.png)
then I click the edit panel and it still doesn't show anything other than temp_1
![Screen Shot 2020-07-17 at 4 34 51 PM](https://user-images.githubusercontent.com/436888/87838350-9e2be280-c84b-11ea-8632-6a878fbf597c.png)
then if I refresh the panel in that view, it shows al the fields
![Screen Shot 2020-07-17 at 4 34 59 PM](https://user-images.githubusercontent.com/436888/87838352-9f5d0f80-c84b-11ea-99f5-b5a9b2c099c9.png)

**Comment by alex88** [comment]
It seems that the refresh changes the max data points from 1230 to 2128, however there are 2016 data points only for temp_1 so even 2128 shouldn't show all variable anyway right?

**Comment by ryantxu** [comment]
Can you manually bump up the max data points and see if that is your issue?  Click on "Query options" and use 5000 or something bigger

![image](https://user-images.githubusercontent.com/705951/87839384-7b033200-c84f-11ea-85ee-6354a9172a3a.png)

**Comment by alex88** [comment]
@ryantxu yeah with 10k it seems to work, I thought it was per field (since it says per data series), so I'll probably need  to bump that up to a very high value. I'll close the issue for now then

**Comment by ryantxu** [comment]
Let's reopen so we can track here

**Comment by alex88** [comment]
Oh ok so there's an actual problem, thank you!

**Comment by torkelo** [comment]
Max datapoints should only set an interval variable for influxdb, but I cannot are any interval variable in your query so not sure how it can effect the result in any way

**Comment by cgfoed** [comment]
> Max datapoints should only set an interval variable for influxdb, but I cannot see any interval variable in your query so not sure how it can effect the result in any way

How _Max datapoints_ is automatically calculated in the first place I don't know and maybe I'm wrong, as I don't write go, but when looking at https://github.com/grafana/grafana/blob/e03da3760efc9be99c6808fd67c6a3e09ee1f2a5/pkg/tsdb/influxdb/flux/executor.go
it **seems** to me that maxPoints is given to builder and this will limit it's size (?)

```go
func readDataFrames(result *api.QueryTableResult, maxPoints int, maxSeries int) (dr backend.DataResponse) {
	glog.Debug("Reading data frames from query result", "maxPoints", maxPoints, "maxSeries", maxSeries)
	dr = backend.DataResponse{}

	builder := &frameBuilder{
		maxPoints: maxPoints,
		maxSeries: maxSeries,
	}
```

**Comment by davkal** [comment]
Fixed in 7.4.1
